docs/partiql-tests-schema-proposal.md

```diff
@@ -137,32 +164,95 @@ envs::{
   'table2': ...
 }
 
-// environment for test can be specified using one of `envs` in the file:
-env: table1
-// or specified explicitly with a struct
-env: { 'table1': [{a:1}, {a:2}, {a:3}] }
+// environment for test can be specified using one of the `envs` in the namespace/file
+{
+   name: "test using environment symbol",
+   statement: "SELECT * FROM table1",
+   ...
+}
+// or specified explicitly using the inlined `env`
+{
+   name: "test using explicit environement",
+   statement: "SELECT * FROM table1",
+   env: { table1: [{a:1}, {a:2}, {a:3}] }
+   ...
+}
 ```
 
+#### Modeling PartiQL Types in Ion
+PartiQL's type system encompasses Ion's types with a few additional types (e.g. missing, bag, date, time). There are a 
+few approaches we could take to model PartiQL data:
+- Annotations (using Ion)
+- S-expressions (using Ion)
+- PartiQL object notation
+
+Of the above options, we've decided to go with the annotation approach. Values of these additional types will be denoted
+using a `$<partiql_type>` annotation. This will be used for the output result and environments.
+
 ```
-// eval 'success' test
+// bag -- list annotated with $bag
+$bag::[1, 2, 3]
+
+// missing -- null value annotated with $missing
+$missing::null
+
+// date -- string annotated with $date
+$date::'2022-02-22'
+
+// time -- string annotated with $time
+$time::'02:30:59'
+```
+
+---
+
+### Equivalence
+The PartiQL specification mentions some PartiQL statements that could be rewritten using a different PartiQL syntax 
+(e.g. wildcard expressions). A common use case could be to assert that such PartiQL statements evaluate to the same 
+result or have the same plan. Users can specify an equivalence class as follows.
+
+```
+equiv_class::{
+    id: <symbol>,               // identifier that can be referred to in equivalence assertions
+    statements: <list<string>>  // list of equivalent PartiQL statements as strings
+}
+```
+
+Evaluation tests can refer to the equivalence classes specified in the file/namespace and make assertions that the
+statements evaluate to the same result.
+
+```
+// evaluation equivalence test
 {
-    name: <string>,
-    statement: <string>,
-    env: <symbol> | <struct>,
-    options: <list<symbol>>,
+    // same fields for evaluation success test
+    ...
     assert: {
-        result: <ion>
-    },
+        result: EvaluationSuccess,
+        output: <ion>,
+        equiv_class: symbol     // identifier to equivalence class
+    }
 }
+```
 
-// eval 'fail' test
+As a simple example, the following would be how to write an evaluation equivalence test:
+```
+// equivalence class definition
+equiv_class::{
+    id: ten,
+    statements: [
+        "5 * 2",
+        "20 / 2",
+        "1 + 2 + 3 + 4",
+    ]
+}
+
+// evaluation test with equivalence class assertion
 {
-    name: <string>,
-    statement: <string>,
-    env: <symbol> | <struct>,
-    options: <list<symbol>>,
+    name: "equivalence class test sample",
+    statement: "10",
```

Comment:
I am inclined to have "statement" reference the equivalence class. The logic being
- If `statement` is a string, it's PartiQL statement
- If `statement` is a symbol, it's referencing an equivalence class of statements

I suggest this because each equivalent query will replace "statement" when evaluated. I'd rather have them placed where they are "used" rather than in the assertion block